source/common/http/conn_manager_impl.cc

```diff
@@ -1569,8 +1579,10 @@ void ConnectionManagerImpl::ActiveStream::sendLocalReply(
   // The BadRequest error code indicates there has been a messaging error.
   if (Runtime::runtimeFeatureEnabled(
           "envoy.reloadable_features.hcm_stream_error_on_invalid_message") &&
-      !connection_manager_.config_.streamErrorOnInvalidHttpMessaging() &&
-      code == Http::Code::BadRequest && connection_manager_.codec_->protocol() < Protocol::Http2) {
+      code == Http::Code::BadRequest && connection_manager_.codec_->protocol() < Protocol::Http2 &&
+      !streamErrorOnInvalidHttpMessage(
```

Comment:
Do we need to runtime guard this twice? Can we remove the first if condition? Also, I haven't been tracking all of these changes, and you probably already discussed this with Alyssa, but is this the only place this needs to be checked? I would have naively assumed there are more places this is needed but maybe not?

Reply:
The runtime guard here is needed because it affects HTTP/1.1 specifically, and the other place where it's checked affects HTTP/2 instead. As far as I understand, prior to https://github.com/envoyproxy/envoy/pull/11714, we were sending errors in two places (codec and HCM), however after that change we now send errors only in the HCM. Not sure if we should check in the other code path as well (see https://github.com/envoyproxy/envoy/pull/11884#issuecomment-661098812).